job-server/src/spark.jobserver/JobInfoActor.scala

```diff
@@ -33,18 +33,19 @@ class JobInfoActor(jobDao: JobDAO, contextSupervisor: ActorRef) extends Instrume
 
   override def wrappedReceive: Receive = {
     case GetJobStatuses(limit) =>
-      sender ! jobDao.getJobInfos(limit.get)
+      val jobInfos = Await.result(jobDao.getJobInfos(limit.get), 60 seconds)
+      sender ! jobInfos
 
     case GetJobStatus(jobId) =>
       val jobInfo = Await.result(jobDao.getJobInfo(jobId), 60 seconds)
```

Comment:
Sorry to put this on you @noorul but please get rid of all other Await.result from this class, and every other class that uses jobDao.  Everything should be Future callbacks, no timeouts.

    jobDao.getJobInfo(jobId).collect {
      case Some(jobId) => originator ! jobId
      case None            => originator ! NoSuchJobId
    }